### PR TITLE
gh-120326: Include ``intrin.h`` on Windows

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-120326

### CODE REVIEW

**Comment by Eclips4** [comment]
@cdgriffith can you confirm that this resolves the issue?

**Comment by cdgriffith** [comment]
@Eclips4 The build completes, and produces a 3.14 executable (assuming this will also be merged into 3.13). 

Thank you for fast fix!

File: Include/object.h [hunk 1]
```diff
@@ -4,6 +4,9 @@
 extern "C" {
 #endif
 
+#if defined(_MSC_VER)
+#include <intrin.h>
```
Comment (vstinner) [inline Include/object.h]: For readability:

```suggestion
#  include <intrin.h>
```

File: Misc/NEWS.d/next/Build/2024-06-11-00-38-05.gh-issue-120326.JHSDF1.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fix build error on Windows when ``--disable-gil`` and ``--experimental-jit``
+options are combined.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Build/2024-06-11-00-38-05.gh-issue-120326.JHSDF1.rst]: I prefer to mention the "domain" first, so the reader can quickly skip if it's unrelated to their use cases:

```suggestion
On Windows, fix build error when ``--disable-gil`` and ``--experimental-jit``
options are combined.
```

**Review by vstinner — APPROVED** [review]
LGTM, I just left minor suggestions.

**Comment by Eclips4** [comment]
Thanks for the review, Victor.

**Review by vstinner — APPROVED** [review]
LGTM

**Review by vstinner — COMMENTED** [review]
Oh wait, now I recall vaguely a C++20 module issue if the include is done inside a `extern "C" { ... }`. Would it be possible to move the include to Python.h, and add a comment explaining why it's needed, mention at least one function which is used?

Also, is this include only needed for Free Threading, or for JIT compiler, or both?

**Comment by vstinner** [comment]
Thanks, I enabled auto-merge.

For example, I see that _Py_ThreadId() uses:

```
#if defined(_MSC_VER) && defined(_M_X64)                           
    tid = __readgsqword(48);
```

**Comment by Eclips4** [comment]
@vstinner I've disabled automerge due to failure of "Tests / Windows (free-threading) / build and test (x86) (pull_request)" job (see https://github.com/python/cpython/actions/runs/9468440435/attempts/1?pr=120329) . I've re=run it and seems it will pass.

I don't know if it is related to this pull request, but it is definitely something strange, and I am not sure we can merge it now. However, I cannot reproduce the failure from the job locally on my Windows setup :(

**Comment by vstinner** [comment]
> @vstinner I've disabled automerge due to failure of "Tests / Windows (free-threading) / build and test (x86) (pull_request)" job (see https://github.com/python/cpython/actions/runs/9468440435/attempts/1?pr=120329) . I've re-run it and seems it will pass (UPD: Yes, it is).

I don't see how an include can trigger this issue. It's strange since the job passed when re-run.

**Comment by vstinner** [comment]
Oh, the x86 job failed again on the commit checks: https://github.com/python/cpython/actions/runs/9470417218/job/26091279722

```
Run .\python.bat -m test.pythoninfo
Running Debug|Win32 interpreter...
Error: Process completed with exit code 1.
```

**Comment by vstinner** [comment]
I built Python with Free Threading on Windows for x86: I failed to reproduce the issue.

I ran `PCbuild\build.bat -e -d -p x86 --disable-gil` to build Python, and then I ran `python.bat -m test.pythoninfo`.

**Comment by vstinner** [comment]
Windows x86 Free Threading failed 2x in a row, and then succeeded 5x in a row. This bug is strange. I don't know how to debug it since I cannot reproduce it locally.

**Pull request merged**